lib/temporal/connection/serializer/continue_as_new.rb

```diff
@@ -31,6 +32,12 @@ def serialize_headers(headers)
 
           Temporal::Api::Common::V1::Header.new(fields: object.headers)
         end
+
+        def serialize_memo(memo)
+          return unless memo
+
+          Temporal::Api::Common::V1::Memo.new(fields: object.memo)
```

Comment:
I'm assuming we need to call `to_payload_map` here to properly serialise it. Also we should use the local `memo` variable, not `object.memo`

Reply:
I'll see if I can cover this in some tests, and clean up the mixed instance/local variable pattern I blindly copied from headers.

Reply:
This should now be fixed for all memos. I'll do the same for headers in a separate PR.